core/domain/exp_jobs_one_off.py

```diff
@@ -506,3 +506,45 @@ def map(exp_summary_model):
     def reduce(key, values):
         """Implements the reduce function for this job."""
         yield (key, len(values))
+
+
+class RegenerateMissingExpCommitLogModels(jobs.BaseMapReduceOneOffJobManager):
+    """Job that regenerates missing commit log models for an exploration."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        rights_model = exp_models.ExplorationRightsModel.get_by_id(item.id)
+
+        for version in python_utils.RANGE(1, item.version + 1):
+            commit_log_model = (
+                exp_models.ExplorationCommitLogEntryModel.get_by_id(
+                    'exploration-%s-%s' % (item.id, version)))
+            if commit_log_model is None:
+                metadata_model = (
+                    exp_models.ExplorationSnapshotMetadataModel.get_by_id(
+                        '%s-%s' % (item.id, version)))
+                commit_log_model = (
+                    exp_models.ExplorationCommitLogEntryModel.create(
```

Comment:
I suggest that, to check this, you write a temporary function (called from a button in the admin tab) that, when the button is clicked, creates these commit log models (using this code) and compares them with the ones that already exist, to make sure that what you're regenerating here will be identical to what the correct version should be. We can do 2-3 spot-checks to validate. This validation is important because there's already a mistake here (see comment below) and if we populate the models with the wrong stuff then cleanup is going to become hard since they're indistinguishable from the correct ones.

Reply:
Want to confirm thing: we cannot compare date time since that won't be exact same, should that be compared with some time diff allowed?

Reply:
Yup! That makes sense.

Reply:
Done. 

Screenshots for verification:

<img width="871" alt="Screenshot 2020-10-01 at 3 29 20 AM" src="https://user-images.githubusercontent.com/15226041/94745630-b4e8cc80-0398-11eb-87e7-5cf4bb26480d.png">
<img width="858" alt="Screenshot 2020-10-01 at 3 29 31 AM" src="https://user-images.githubusercontent.com/15226041/94745638-b7e3bd00-0398-11eb-87b5-fe0915d68e36.png">


Want to check regarding date time, I added a delta of 1 minute just to be a bit on lenient side. Should it be decreased?